cmd/kubeadm/app/apis/kubeadm/v1beta4/types.go

```diff
@@ -282,6 +286,10 @@ type LocalEtcd struct {
 	// +optional
 	ExtraArgs map[string]string `json:"extraArgs,omitempty"`
 
+	// ExtraEnvs is an extra set of environment variables to pass to the control plane component.
+	// +optional
+	ExtraEnvs []corev1.EnvVar `json:"extraEnvs,omitempty"`
```

Comment:
As below, the `ExtraEnvs` will overwrite the host `ProxyEnv` on purpose.

Nit: to mention it in the comments?

Reply:
well, what we have here is corev1.EnvVar, which means users can actually use Value but also ValueFrom:
https://pkg.go.dev/k8s.io/api/core/v1#EnvVar

and they can do things like this:
https://kubernetes.io/docs/tasks/inject-data-application/environment-variable-expose-pod-information/#use-pod-fields-as-values-for-environment-variables

so technically our ExtraEnvs will support the downward API.

Reply:
> As below, the `ExtraEnvs` will overwrite the host `ProxyEnv` on purpose if the same(only for `*_proxy`).
> 
> Nit: to mention it in the comments?

hmm, that's a good idea, in case users ask.

@chendave we can use this, WDYT?

>> // ExtraEnvs is an extra set of environment variables to pass to the control plane component.
>> // Environment variables passed using ExtraEnvs will override any existing environment variables,
>> // or *_proxy environment variables that kubeadm adds by default.